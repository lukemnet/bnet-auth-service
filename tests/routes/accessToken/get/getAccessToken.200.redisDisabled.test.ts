import fastify from 'fastify';
import fastifyRedis from 'fastify-redis-mock';
import server from '../../../../src/index';
import getConfig from '../../../helper';

describe('/accessToken/get 200 (Redis disabled)', () => {
  const fastifyServer = fastify();

  beforeAll(async () => {
    fastifyServer.register(fastifyRedis, {
      host: '127.0.0.1',
      port: '6379',
      password: '',
      enableReadyCheck: true,
      dropBufferSupport: false,
    });
    fastifyServer.register(server, getConfig(false));
  });

  afterEach(() => {
    fastifyServer.close();
  });

  it('returns 200', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/get', });
    expect(res.statusCode).toBe(200);
  });

  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/get', });
    expect(JSON.parse(res.payload)).toEqual({"status":200,"data":{ accessToken: "sample access token" }});
  });
});