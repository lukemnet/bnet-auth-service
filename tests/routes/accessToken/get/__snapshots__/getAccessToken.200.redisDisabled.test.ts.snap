// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/accessToken/get 200 (Redis disabled) returns correct response 1`] = `"{\\"status\\":200,\\"data\\":{\\"accessToken\\":\\"sample access token\\"}}"`;
